Tribler/Core/Utilities/utilities.py

```diff
@@ -75,35 +79,93 @@ def validTorrentFile(metainfo):
                 logger.info("address port not int, but %s", type(address[1]))
             else:
                 valid_initial_peers.append(address)
+    return valid_initial_peers
 
-        metainfo['initial peers'] = valid_initial_peers
 
-    info = metainfo['info']
-    if not isinstance(info, DictType):
-        raise ValueError('info not dict')
+def validate_url_list(metainfo):
+    """
+    Validate the list of URLs in the metainfo if such list exists and remove wrong URLs.
 
-    if 'root hash' in info:
-        infokeys = ['name', 'piece length', 'root hash']
-    else:
-        infokeys = ['name', 'piece length', 'pieces']
-    for key in infokeys:
-        if key not in info:
-            raise ValueError('info misses key ' + key)
-    name = info['name']
-    if not isinstance(name, StringType):
-        raise ValueError('info name is not string but ' + repr(type(name)))
-    pl = info['piece length']
-    if not isinstance(pl, IntType) and not isinstance(pl, LongType):
-        raise ValueError('info piece size is not int, but ' + repr(type(pl)))
-    if 'root hash' in info:
-        rh = info['root hash']
-        if not isinstance(rh, StringType) or len(rh) != 20:
-            raise ValueError('info roothash is not 20-byte string')
-    else:
-        p = info['pieces']
-        if not isinstance(p, StringType) or len(p) % 20 != 0:
-            raise ValueError('info pieces is not multiple of 20 bytes')
+    First checks whether a URL-list exists, after which it is checked whether the metainfo
+    would be able to use HTTP seeding. This is not the case if there are multiple files specified.
+
+    A warning is logged if:
```

Comment:
Although this is specified in the `return` line of the docstring, I've made sure to make that more clear.